### DISCUSSION TITLE
How to install dynamic lib

### DISCUSSION THREAD

**Discussion by zhanghenan-dev** [keplerproject/luarocks#1442] [Q&A] [answered]
I have a .rockspec file,  just want to change installing a dynamic lib to another path. The default path is /usr/lib64/lua/5.1/ , but I want to install in /etc/nginx/lsqlite3/lib, please tell me how to finish it ? Thank you .

Origin .rockspec
-----------------------------------------------
package = "lsqlite3"
version = "0.9.5-1"
source = {
    url = "http://lua.sqlite.org/index.cgi/zip/lsqlite3_fsl09y.zip?uuid=fsl_9y",
    file = "lsqlite3_fsl09y.zip"
}
description = {
    summary = "A binding for Lua to the SQLite3 database library",
    detailed = [[
        lsqlite3 is a thin wrapper around the public domain SQLite3 database engine. SQLite3 is
        dynamically linked to lsqlite3. The statically linked alternative is lsqlite3complete.
        The lsqlite3 module supports the creation and manipulation of SQLite3 databases.
        Most sqlite3 functions are called via an object-oriented interface to either
        database or SQL statement objects.
    ]],
    license = "MIT",
    homepage = "http://lua.sqlite.org/"
}
dependencies = {
    "lua >= 5.1, < 5.5"
}
external_dependencies = {
    SQLITE = {
        header = "sqlite3.h"
    }
}
build = {
    type = "builtin",
    modules = {
        lsqlite3 = {
            sources = { "lsqlite3.c" },
            defines = {'LSQLITE_VERSION="0.9.5"'},
            libraries = { "sqlite3" },
            incdirs = { "$(SQLITE_INCDIR)" },
            libdirs = { "$(SQLITE_LIBDIR)" }
        },
    },
    copy_directories = { 'doc', 'examples' },
}
--------------------------------------------------------

After change:
--------------------------------------------------------
......
build = {
    type = "builtin",
    modules = {
        lsqlite3 = {
            sources = { "lsqlite3.c" },
            defines = {'LSQLITE_VERSION="0.9.5"'},
            libraries = { "sqlite3" },
            incdirs = { "$(SQLITE_INCDIR)" },
            libdirs = { "$(SQLITE_LIBDIR)" }
        },
    },
    copy_directories = { 'doc', 'examples' },
    install = {
        lib = {
            ["/etc/nginx/lsqlite3/lib/lsqlite3"] = { "lsqlite3.so" },
        },
    },
}
-----------------------------------------------------------

When compile, error is :
luarocks make ./lsqlite3-0.9.5-1.rockspec SQLITE_DIR=/etc/nginx/lsqlite3 SQLITE_INCDIR=/etc/nginx/lsqlite3/include LUA_INCDIR=/etc/nginx/luajit/include/luajit-2.1 PREFIX=/etc/nginx/lsqlite3
gcc -O2 -fPIC -I/etc/nginx/luajit/include/luajit-2.1 -c lsqlite3.c -o lsqlite3.o -DLSQLITE_VERSION="0.9.5" -I/etc/nginx/lsqlite3/include
gcc -shared -o lsqlite3.so -L/usr/lib64/lua/5.1 lsqlite3.o -L/etc/nginx/lsqlite3/lib -Wl,-rpath,/etc/nginx/lsqlite3/lib: -lsqlite3

Error: LuaRocks 2.3.0 bug (please report at https://github.com/keplerproject/luarocks/issues).
/usr/lib64/lua/5.1/luarocks/dir.lua:14: assertion failed!
stack traceback:
        [C]: in function 'assert'
        /usr/lib64/lua/5.1/luarocks/dir.lua:14: in function 'base_name'
        /usr/lib64/lua/5.1/luarocks/build.lua:64: in function 'install_files'
        /usr/lib64/lua/5.1/luarocks/build.lua:273: in function 'build_rockspec'
        /usr/lib64/lua/5.1/luarocks/make.lua:82: in function </usr/lib64/lua/5.1/luarocks/make.lua:51>
        (tail call): ?
        [C]: in function 'xpcall'
        /usr/lib64/lua/5.1/luarocks/command_line.lua:207: in function 'run_command'
        /usr/bin/luarocks:33: in main chunk
        [C]: ?
make: *** [install] Error 99

I  don't know how to write the install path?

**Comment by alerque** [comment, marked as answer]
LuaRocks is not made for installing things outside of a Lua module tree to be consumed by Lua apps. You should probably include some other build & packaging system if that's the result you want.

**Reply by zhanghenan-dev** [reply]
I got it. Thank you !
